### DISCUSSION TITLE
Blog 7: Architektur

### DISCUSSION THREAD

**Discussion by Max-Rohrhirsch** [DHBW-UMOC/UMOC#11] [Announcements]
# Qualitätsbaum
Für unser Projekt haben wir uns für eine Mischung an Servicebasierter, Eventbasierter und Schichtenarchitektur entschieden. Hierfür haben wir folgenden Qualitätsbaum entworfen:
![image](https://github.com/user-attachments/assets/878944e9-b585-4e21-8d89-1ed6d59e5d02)

### [A1] Latenz
- Antwortzeit für Nachrichten
- Verbindungsaufbauzeit
### [A2] Durchsatz
- Anzahl der Nachrichten pro Sekunde
- Maximale gleichzeitige Verbindungen
### [A3] Ressourcennutzung
- CPU-Auslastung
- Speicherverbrauch

### [B1] Authentifizierung
- Methoden der Benutzeranmeldung (z.B. OAuth, Zwei-Faktor)
### [B2] Vertraulichkeit
- Verschlüsselung (z.B. AES für gespeicherte Nachrichten)
### [B3] Zugriffskontrolle
- Rollenbasierte Zugriffskontrolle
- Berechtigungen für verschiedene Benutzer

### [C1] Benutzeroberfläche
- Klarheit der Navigation
- Konsistenz der Gestaltung
### [C2] Reaktionsfähigkeit
- Ladezeiten der Benutzeroberfläche
- Zeit bis zur ersten Aktion (z.B. Nachrichten senden)
### [C3] Hilfestellungen
- Tooltips
- FAQs und Onboarding-Prozesse

### [D1] Verfügbarkeit
- Uptime: Prozentuale Verfügbarkeit des Systems über einen definierten Zeitraum (z.B. 99,9%).
- Fehlerbehandlung: Mechanismen zur automatischen Wiederherstellung nach Serverausfällen.
### [D2] Fehlertoleranz
- Backup-Mechanismen: Maßnahmen zur Sicherung von Daten und Kontinuität bei Systemausfällen.
- Redundanz: Einsatz von redundanten Komponenten (z.B. Server, Datenbanken) zur Sicherstellung der Betriebskontinuität.
### [D3] Wiederherstellbarkeit
- Backup-Frequenz: Häufigkeit der Datensicherungen zur Minimierung von Datenverlust.
- Wiederherstellungszeit: Zeit, die benötigt wird, um nach einem Ausfall den Normalbetrieb wiederherzustellen (RTO – Recovery Time Objective).

**Comment by ktoffel999** [comment]
Der Qualitätsbaum bietet einen umfassenden und präzisen Rahmen zur Bewertung und Verbesserung wichtiger Systemeigenschaften. Er konzentriert sich auf essentielle Performance-Aspekte, gewährleistet Sicherheit, fördert Benutzerfreundlichkeit und sichert Zuverlässigkeit. Insgesamt ist der Ansatz sehr zielgerichtet und deckt kritische Qualitätsmerkmale effektiv ab.
MfG das Jump and Run Team

**Comment by bunki1224** [comment]
Euer Qualitätsbaum sieht super strukturiert aus! Die Aufteilung in Qualitätsmerkmale, verfeinerte Merkmale und Szenarien ist logisch und gut nachvollziehbar.

- Performance: Die Unterpunkte (Latenz, Durchsatz, Ressourcennutzung) decken die wichtigsten Metriken ab. Vielleicht könntet ihr noch „Netzwerklatenz“ ergänzen, falls relevant.

- Sicherheit: Schön, dass Authentifizierung und Zugriffskontrolle dabei sind! Verschlüsselung während der Übertragung (z. B. TLS) könnte noch ein Thema sein.

- Usability: Gut, dass ihr auch Hilfestellungen wie Tooltips und Onboarding-Prozesse mit einplant. Nutzerfeedback könnte hier noch ein Plus sein.

- Zuverlässigkeit: Verfügbarkeit, Fehlertoleranz und Wiederherstellbarkeit sind super abgedeckt. Monitoring-Systeme könnten noch helfen, Probleme frühzeitig zu erkennen.

Insgesamt ein klarer und durchdachter Plan 🚀
>Euer TripAhead Team (JB)
